pkg/controllers/systemd-etcd.go

```diff
@@ -0,0 +1,153 @@
+func (s *ExtEtcdService) Dependencies() []string { return []string{} }
+
+func (s *ExtEtcdService) configure(cfg *config.MicroshiftConfig) {
+	s.NodeName = cfg.NodeName
+
+	s.DataDir = filepath.Join(microshiftDataDir, s.Name())
+	certsDir := cryptomaterial.CertsDirectory(microshiftDataDir)
+	etcdServingCertDir := cryptomaterial.EtcdServingCertDir(certsDir)
+	etcdPeerCertDir := cryptomaterial.EtcdPeerCertDir(certsDir)
+	etcdSignerCertPath := cryptomaterial.CACertPath(cryptomaterial.EtcdSignerDir(certsDir))
+
+	s.CertFile = cryptomaterial.PeerCertPath(etcdServingCertDir)
+	s.KeyFile = cryptomaterial.PeerKeyPath(etcdServingCertDir)
+	s.TrustedCAFile = etcdSignerCertPath
+
+	s.PeerCertFile = cryptomaterial.PeerCertPath(etcdPeerCertDir)
+	s.PeerKeyFile = cryptomaterial.PeerKeyPath(etcdPeerCertDir)
+	s.PeerTrustedCAFile = etcdSignerCertPath
+
+	s.InitialUrl = fmt.Sprintf("https://%s:2380", cfg.NodeIP)
+	s.AdvertiseClientUrls = fmt.Sprintf("https://%v:2379", cfg.NodeIP)
+	s.ListenClientUrls = fmt.Sprintf("https://127.0.0.1:2381,https://%v:2381", cfg.NodeIP)
+	s.ListenPeerUrls = fmt.Sprintf("https://%v:2380", cfg.NodeIP)
+	s.ListenMetricsUrls = "127.0.0.1:2381"
+}
+
+func (s *ExtEtcdService) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+	defer close(stopped)
+
+	// Check to see if we should run as a systemd run or directly as a binary.
+	runningAsSvc := false
+	if os.Getenv("INVOCATION_ID") != "" {
+		runningAsSvc = true
+	}
+
+	execPath, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("%v failed to get exec path: %v", s.Name(), err)
+	}
+
+	var etcdPath string
+	var args []string
+	// This is a temporary measure to run embedded etcd with the transient systemd unit.
+	// TODO remove this once we have the etcd binary laid down.
+	if os.Getenv("EXTERNAL_ETCD") == "" {
+		etcdPath = fmt.Sprintf("%v", filepath.Join(execPath, "microshift"))
+		args = []string{"etcd"}
+	} else {
+		// Create the path to the etcd executable.
+		etcdPath = filepath.Join(filepath.Dir(execPath), "etcd")
+
+		// These are the shared arguments for both etcd invocations
+		args = []string{
+			"--name=" + s.NodeName,
+			"--data-dir=" + s.DataDir,
+			"--logger=zap",
+			"--experimental-initial-corrupt-check=true",
+			"--initial-advertise-peer-urls=" + s.InitialUrl,
+			"--cipher-suites=" + strings.Join(tlsCipherSuites, ","),
+			"--cert-file=" + s.CertFile,
+			"--key-file=" + s.KeyFile,
+			"--trusted-ca-file=" + s.TrustedCAFile,
+			"--client-cert-auth=true",
+			// TODO - we might not need peers since we're single node
+			"--peer-cert-file=" + s.PeerCertFile,
+			"--peer-key-file=" + s.PeerKeyFile,
+			"--peer-trusted-ca-file=" + s.PeerTrustedCAFile,
+			"--peer-client-cert-auth=true",
+			"--advertise-client-urls=" + s.AdvertiseClientUrls,
+			"--listen-client-urls=" + s.ListenClientUrls,
+			"--metrics=extensive",
+			"--listen-metrics-urls=" + s.ListenMetricsUrls,
```

Comment:
So this looks like a foundation for situation when we use etcd binary directly (with no microshift-etcd, shims, etc.). Do we actually need that right now? I think this might be confusing when reading code

Reply:
We don't need it right now, which is why I put it under a flag so we could start manually testing with the binary.